### PR TITLE
Scaffold books api with basic, stubbed get endpoint

### PR DESCRIPTION
**Pull request by kingscott**
This replaces #2, there was an issue with my commit from another machine 👀

### CODE REVIEW

File: pkg/v1/endpoints/books/request.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import (
+    "github.com/King-Rullmann-Book-Club/books-api/pkg/v1/service/books"
+)
+
+type GetBookRequest struct {
+    ID string
+}
+
+type GetBookResponse struct {
+    Book books.Book `json:"book,omitempty"`
```
Comment (erullmann) [inline pkg/v1/endpoints/books/request.go:12]: Not sure if this is intentional, but having this be a go object means that the response has non-standard json casing.

```
$ curl localhost:8080/books/1
{"book":{"Name":"Perdido Street Station"}}
```
Note the capitalization of `Name`.

Could this be solved by changing the interface of Book like so?
```
type Book struct {
    Name string `json:"name"`
}
```

**Review by erullmann — APPROVED** [review]

File: pkg/v1/endpoints/books/request.go (hunk 1 above)
Comment (kingscott) [reply pkg/v1/endpoints/books/request.go:12]: Yes it can. I'll update.

File: cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    //		httpAddr = flag.String("http.addr", ":8080", "HTTP listen address")
+    //	)
+    //	flag.Parse()
+
+    var logger log.Logger
+    {
+        logger = log.NewLogfmtLogger(os.Stderr)
+        logger = log.With(logger, "ts", log.DefaultTimestampUTC)
+        logger = log.With(logger, "caller", log.DefaultCaller)
+    }
```
Comment (erullmann) [inline cmd/main.go:27]: Could you explain this a bit? So far as I can tell, this is a scope wrapping some initalization functions, but it has no impact on the defintiion of logger. It is the same as 

```
var logger log.Logger
logger = log.NewLogfmtLogger(os.Stderr)
logger = log.With(logger, "ts", log.DefaultTimestampUTC)
logger = log.With(logger, "caller", log.DefaultCaller)
```

Is that right? Is this a stylistic choice or is there some effect I'm missing?

**Review by erullmann — APPROVED** [review]
LGTM, the channel stuff took me a sec.

File: cmd/main.go (hunk 2 above)
Comment (kingscott) [reply cmd/main.go:27]: I don't believe they are exactly the same. What happens inside the brace brackets, stays within the brace brackets :sunglasses: It's a scoping thing. It's common in Go to define a variable, in this case `logger` that is going to receive some manipulation, to define the variable using `var` and then modify the variable within a close scope. 

You'll see this with context as well:

```go
var ctx context.Context
{
  ctx = context.WithValue(context.Background(), "foo", "bar")
  ctx = context.WithValue(ctx, "hello", "world")
}
```

**Pull request merged**